core/src/banking_stage/qos_service.rs

```diff
@@ -146,6 +146,31 @@ impl QosService {
         }
     }
 
+    /// Record unadjusted costs for committed transactions.
+    /// This should be called instead of `update_costs` when the feature is
+    /// activated to no longer update costs.
+    pub fn record_unadjusted_costs<'a>(
+        transaction_cost_results: impl Iterator<Item = &'a transaction::Result<TransactionCost>>,
+        transaction_committed_status: Option<&Vec<CommitTransactionDetails>>,
+        bank: &Bank,
+    ) {
+        if let Some(transaction_committed_status) = transaction_committed_status {
+            let mut cost_tracker = bank.write_cost_tracker().unwrap();
```

Comment:
nit: avoid write lock when transaction_cost_results or transaction_committed_status is empty?